fuel-core/src/schema/coin.rs

```diff
@@ -111,105 +110,39 @@ impl CoinQuery {
+                    if let (Ok((_, coin)), Some(filter_asset_id)) =
+                        (&result, &filter.asset_id)
                     {
-                        return Err(anyhow!("Wrong argument combination"))
+                        if coin.1.asset_id != filter_asset_id.0 {
+                            return None
+                        }
                     }
 
-                    let after = after.map(fuel_tx::UtxoId::from);
-                    let before = before.map(fuel_tx::UtxoId::from);
+                    Some(result)
+                });
 
-                    let start;
-                    let end;
-
-                    if direction == IterDirection::Forward {
-                        start = after;
-                        end = before;
-                    } else {
-                        start = before;
-                        end = after;
-                    }
-
-                    let owner: fuel_tx::Address = filter.owner.into();
-
-                    let mut coin_ids = db.owned_coins_ids(&owner, start, Some(direction));
-                    let mut started = None;
-                    if start.is_some() {
-                        // skip initial result
-                        started = coin_ids.next();
-                    }
-
-                    // take desired amount of results
-                    let coins = coin_ids
-                        .take_while(|r| {
-                            // take until we've reached the end
-                            if let (Ok(t), Some(end)) = (r, end.as_ref()) {
-                                if *t == *end {
-                                    return false
-                                }
-                            }
-                            true
-                        })
-                        .take(records_to_fetch);
-                    let mut coins: Vec<fuel_tx::UtxoId> = coins.try_collect()?;
-                    if direction == IterDirection::Reverse {
-                        coins.reverse();
-                    }
-
-                    // TODO: do a batch get instead
-                    let coins: Vec<Coin> = coins
-                        .into_iter()
-                        .map(|id| {
-                            db.storage::<Coins>()
-                                .get(&id)
-                                .transpose()
-                                .ok_or(not_found!(Coins))?
-                                .map(|coin| Coin(id, coin.into_owned()))
-                        })
-                        .try_collect()?;
-
-                    // filter coins by asset ID
-                    let mut coins = coins;
-                    if let Some(asset_id) = filter.asset_id {
-                        coins.retain(|coin| coin.1.asset_id == asset_id.0);
-                    }
-
-                    // filter coins by status
-                    coins.retain(|coin| coin.1.status == CoinStatusModel::Unspent);
```

Comment:
I removed this filter because, on the SDK side, we use it to get all coins(spent and unspent).